core-tests/shared/src/test/scala/zio/ChunkSpec.scala

```diff
@@ -565,6 +565,11 @@ object ChunkSpec extends ZIOBaseSpec {
     zio.test.test("Tags.fromValue is safe on Scala.is") {
       val _ = Chunk(1, 128)
       assertCompletes
+    },
+    testM("chunks can be constructed from heterogeneous collections") {
+      check(Gen.listOf(Gen.oneOf(Gen.anyInt, Gen.anyString))) { as =>
```

Comment:
Isn't `String` non-primitive in the sense that it is a subtype fo `AnyRef` as opposed to `AnyVal`? Or were you thinking of something different?

Reply:
The bug we're fixing here existed because heterogeneous collections were not handled correctly. It's an implementation detail that this only happened in the presence of `AnyVal`s, because there are no optimised implementations for `AnyRef`s. In the future we might for example introduce a `Chunk` optimised for `String`s. Then we could run into the same type of bug again. I was thinking of generating `Option`s, or case classes sharing trait.

Reply:
Sounds good. Will add.